### PR TITLE
Feature: navigation components

### PR DESCRIPTION
**Pull request by LuisMaja**
Within this feature request we will add a dynamic component for the icons and a item component for the navigation layout #8 #10

### CODE REVIEW

File: src/components/DynamicIcon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+interface IconProps {
+  type: string;
+}
+
+// This dynamically imports an icon component using the (reacts) lazy function.
+// Meaning it will happen asynchronously, which can improve performance by splitting the code into smaller chunks.
+
+const Icon: React.FC<IconProps> = ({ type }) => {
+  const IconComponent = React.lazy(() => import(`../icons/${type}`));
```
Comment (cedricve) [inline src/components/DynamicIcon/index.tsx:15]: Is the issue resolved with the "not found" error?

Screenshot 2024-05-27 at 17 08 41

File: src/components/DynamicIcon/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  title: "Components/Dynamic Icon",
+  component: Icon,
+  tags: ["autodocs"],
+};
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
```
Comment (cedricve) [inline src/components/DynamicIcon/index.stories.tsx:14]: @LuisMaja  currently there is only 1 icon. Could we visualise all of them with separate stories?

File: src/layouts/Navigation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import Text from "../../components/Text";
+import IconDashboard from "../../components/icons/dashboard";
+import IconHistory from "../../components/icons/history";
+import IconLogOut from "../../components/icons/logout";
+import IconNotification from "../../components/icons/notification";
+import IconProfile from "../../components/icons/profile";
+
+const Navigation: React.FC = () => {
+  return (
+    <Box className="border max-h-screen m-0 p-0">
```
Comment (cedricve) [inline src/layouts/Navigation/index.tsx:15]: Do we have a separate navigation item now?

**Review by cedricve — CHANGES_REQUESTED** [review]
Not sure if this is the latest version, but requires more changes, see comments

File: src/components/DynamicIcon/index.tsx (hunk 1 above)
Comment (LuisMaja) [reply src/components/DynamicIcon/index.tsx]: This has been fixed!

File: src/layouts/Navigation/index.tsx (hunk 3 above)
Comment (LuisMaja) [reply src/layouts/Navigation/index.tsx]: The navigation item has been made and implemented into layout

File: src/components/DynamicIcon/index.stories.tsx (hunk 2 above)
Comment (LuisMaja) [reply src/components/DynamicIcon/index.stories.tsx:14]: a story has been created to visualise all icons individually

Comment (cedricve) [reply src/components/DynamicIcon/index.stories.tsx:14]: Looking good!! <3

**Comment by cedricve** [comment]
a separate PR will be created

**Pull request closed without merging**